### PR TITLE
Fixes handling of nan and inf, which previously would cause "Bad Request"

### PR DESCRIPTION
**Pull request by j316chuck**
# What does this PR do?
Fixes handling of nan and inf, which previously would cause "Bad Request". 

@dakinggg merged the composer dev main PR, rewrote the if else condition to be more comprehensive, and added some unit tests in the original PR [here](https://github.com/mosaicml/composer/pull/2727). Now PR GPU tests have been passing 3/3 times [here](https://github.com/mosaicml/composer/actions/runs/6936555074?pr=2734) 

# What issue(s) does this change relate to?
Closes [GRT-2581](https://databricks.atlassian.net/browse/GRT-2581)

# Before submitting
- [x] Have you read the [contributor guidelines](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md)?
- [x] Was this change discussed/approved in a GitHub issue first? It is much more likely to be merged if so.
- [x] Did you update any related docs and document your change?
- [x] Did you update any related tests and add any new tests related to your change? (see [testing](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md#running-tests))
- [x] Did you run the tests locally to make sure they pass?
- [x] Did you run `pre-commit` on your change? (see the `pre-commit` section of [prerequisites](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md#prerequisites))

### CODE REVIEW

**Comment by j316chuck** [comment]
@dakinggg do we still need this after https://github.com/mosaicml/composer/commit/92d983a3d617975dc70364a564a8f22758e92fc0? 

Debugging this GPU flakiness seems hard

**Comment by dakinggg** [comment]
@j316chuck might be ok to just not represent inf and nan...super weird that this causes that test failure though.

**Comment by dakinggg** [comment]
lets see what @mvpatel2000 thinks

File: .github/workflows/pytest-gpu.yaml [hunk 1]
```diff
@@ -83,7 +83,7 @@ jobs:
             REF_ARGS="--pr_number $PR_NUMBER"
           fi
 
-          python .github/mcli/mcli_pytest.py --image '${{ inputs.container }}' --pip_package_name '${{ inputs.composer_package_name }}' --pytest_markers '${{ inputs.pytest-markers }}' --pytest_command '${{ inputs.pytest-command }}' --timeout ${{ inputs.mcloud-timeout }} ${REF_ARGS}
+          python .github/mcli/mcli_pytest.py --image '${{ inputs.container }}' --pip_package_name '${{ inputs.composer_package_name }}' --pytest_markers '${{ inputs.pytest-markers }}' --pytest_command '${{ inputs.pytest-command }}' --timeout ${{ inputs.mcloud-timeout }} ${REF_ARGS} -k "test_resumption"
```
Comment (mvpatel2000) [inline .github/workflows/pytest-gpu.yaml:86]:
```suggestion
          python .github/mcli/mcli_pytest.py --image '${{ inputs.container }}' --pip_package_name '${{ inputs.composer_package_name }}' --pytest_markers '${{ inputs.pytest-markers }}' --pytest_command '${{ inputs.pytest-command }}' --timeout ${{ inputs.mcloud-timeout }} ${REF_ARGS}
```

**Review by mvpatel2000 — COMMENTED** [review]
I'm fine not representing Inf/NaN though I think we should / current solution seems fine

**Comment by j316chuck** [comment]
Okay I'll punt for now since the relative impact of this change is little and the fix seems hard (can't repro consistently locally, need to run 3x on CI for test to fail). I can reopen if this becomes issue down the line.

**Pull request closed without merging**